### PR TITLE
feat(backend): update product API

### PR DESCRIPTION
**Pull request by markjoshuacarmen**
## Ticket/s
[[API] Update Product](https://trello.com/c/OrWeGjL5/34-api-update-product)

## Changes
- add update product API
- modify product controller
- modify product services

## Screenshot

image

## How to test?
   1. Create product 
   2. Use this link
   ```
curl --location 'http://localhost:3001/products/create' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InNhbXBsZSIsInN1YiI6IjY0YzlmZDExMThlNGYyNzRkZTc1YWYxYSIsImlhdCI6MTY5MDk1OTEzNSwiZXhwIjoxNjkxMDQ1NTM1fQ.ZJ1uOuCfEQvCCOi0QmKq3wG_v6d_SRcQB_SkUKZS04o' \
--data '{
 "productID": "product12",
 "productImg": {
     "ImgURL": "https://example.com/image.jpg",
     "ImgAttch": "base64encodedimage"
 },
 "productName": "Product Name",
 "productInfo": "Product Information",
 "productPrice": 199.99,
 "productInventory": 32,
 "productCategory": [
     "sample",
     "sample1",
     "sample2"
 ]
}'
```
   4. Copy the product Id in Db
   5. Copy the product token in login then paste in the Authorization Bearer in PUTrequest
   6. Import cURL for the PUT request.
```
curl --location --request PUT 'http://localhost:3001/products/update/64ca8db9672d65dcba2471e3' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InNhbXBsZSIsInN1YiI6IjY0YzljMThlOGYwODY3NmU3YzgyYzA3NCIsImlhdCI6MTY5MDk0Mzk3OSwiZXhwIjoxNjkxMDMwMzc5fQ.HV-9LG--HZ-73qwY-xX0rkM3wP8ky5Hzu8FQOQ48YzQ' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "change",
    "productImg": {
        "ImgURL": "https://example.com/image.jpg",
        "ImgAttch": "base64encodedimage"
    },
    "productName": "test",
    "productInfo": "Product Information",
    "productPrice": 23,
    "productInventory": 32,
    "productCategory": [
        "sample",
        "sample1",
        "sample2"
    ]
}
'
```
   7. replace <place id here> placeholder with product _id from mongoDB
curl --location 'localhost:3001/products/update/<place id here>'

### CODE REVIEW

**Review by markjoshuacarmen — COMMENTED** [review]
wip(backend): update product API

**Comment by sandrex-SGV** [comment]
- Product update accepts input beginning and/or ending with spaces (see attached img)
- Use the `custom-validations.utils.ts` in the utils directory 

image

**Review by sandrex-SGV — CHANGES_REQUESTED** [review]
Please see comment

https://github.com/dandalandone/POC-EyCommerce/pull/35#issuecomment-1664906622

**Comment by TreshAnn** [comment]
> * Product update accepts input beginning and/or ending with spaces (see attached img)
> * Use the `custom-validations.utils.ts` in the utils directory
> 
> image

same with me. it accepts leading and trailing spaces. As for the Product_Id,  it should not be editable once created

File: apps/api/src/products/products.service.ts [hunk 1]
```diff
@@ -38,4 +38,32 @@ export class ProductsService {
...
+      { _id: id },
+      createProductDto,
+      { new: true },
+    );
+
+    if (!updatedProduct) {
+      throw new NotFoundException('Product not found');
+    }
+
+    console.log(updatedProduct);
```
Comment (TreshAnn) [inline apps/api/src/products/products.service.ts]: remove this
```suggestion
   
```

**Review by TreshAnn — CHANGES_REQUESTED** [review]

File: apps/api/src/products/products.service.ts (hunk 1 above)
Comment (markjoshuacarmen) [reply apps/api/src/products/products.service.ts]: done

**Review by sandrex-SGV — APPROVED** [review]
LGTM!

**Review by TreshAnn — APPROVED** [review]
Working on my end. Thanks, sir!

**Pull request merged**